backend/src/chatroom/chatroom.service.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  Injectable,
+  NotFoundException,
+  ForbiddenException
+} from '@nestjs/common';
+import { CreateChatroomDto, UpdateChatroomDto } from './dto';
+import { PrismaService } from '../prisma/prisma.service';
+import { ChatRoom, ChatRoomUser, Role, RoomType } from '@prisma/client';
+import * as argon from 'argon2';
+
+@Injectable()
+export class ChatroomService {
+  private static DEFAULT_CHATROOM = 'general';
+
+  constructor(private prisma: PrismaService) {}
+  async create(userId: string, dto: CreateChatroomDto) {
+    const snakecaseName = dto.name.toLowerCase().replaceAll(' ', '_');
+    let hash = null;
+
+    if (dto.type === RoomType.PROTECTED && !dto.password) {
+      throw new ForbiddenException('Password is required for protected rooms');
+    }
+
+    if (dto.type === RoomType.PROTECTED && dto.password) {
+      hash = await argon.hash(dto.password);
+    }
+
+    try {
+      const chatroom = await this.prisma.chatRoom.create({
+        data: {
+          name: dto.name,
+          type: dto.type,
+          hash: hash,
+          slug: `chatroom_${snakecaseName}`,
+          users: {
+            create: [
+              {
+                user: { connect: { id: userId } },
+                role: Role.OWNER
+              }
+            ]
+          }
+        }
+      });
+      return chatroom;
+    } catch (e) {
+      throw new NotFoundException('Could not create chatroom');
+    }
+  }
+
+  async findAll() {
+    return await this.prisma.chatRoom.findMany();
+  }
+
+  async findOne(id: string) {
+    return await this.prisma.chatRoom.findUnique({
+      where: { id },
+      include: { users: true }
+    });
+  }
+
+  update(id: string, updateChatroomDto: UpdateChatroomDto) {
+    return `This action updates a #${id} chatroom`;
+  }
+
+  remove(id: string) {
+    return `This action removes a #${id} chatroom`;
+  }
+
+  async findChatroomUsers(chatroomId: string) {
+    return await this.prisma.user.findMany({
+      where: {
+        chatrooms: {
+          some: {
+            chatRoom: { id: chatroomId }
+          }
+        }
+      },
+      select: {
+        id: true,
+        nickname: true,
+        chatrooms: {
+          select: {
+            role: true,
+            chatRoom: false,
+            userId: false
+          },
+          where: {
+            chatRoomId: chatroomId
+          }
+        }
+      }
+    });
+  }
+
+  async join(
+    userId: string,
+    chatroomId: string,
+    password: string
+  ): Promise<ChatRoomUser> {
+    const chatroom = await this.findOne(chatroomId);
+
+    if (!chatroom) {
+      throw new NotFoundException('Chatroom not found');
```

Comment:
It would break every other parts of the code that uses this specific findOne without error handling.

In general, find methods should NOT throw errors unless it's a DB level error